src/plugins/dashboard/public/application/utils/use/use_dashboard_app_state.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import EventEmitter from 'events';
+import { useEffect, useState } from 'react';
+import { cloneDeep } from 'lodash';
+import { map } from 'rxjs/operators';
+import { connectToQueryState, opensearchFilters } from '../../../../../data/public';
+import { migrateLegacyQuery } from '../../lib/migrate_legacy_query';
+import { DashboardServices } from '../../../types';
+
+import { DashboardAppStateContainer } from '../../../types';
+import { migrateAppState, getAppStateDefaults } from '../../lib';
+import { createDashboardAppState } from '../create_dashboard_app_state';
+
+/**
+ * This effect is responsible for instantiating the dashboard app state container,
+ * which is in sync with "_a" url param
+ */
+export const useDashboardAppState = (
+  services: DashboardServices,
+  eventEmitter: EventEmitter,
+  instance: any
```

Comment:
Are we gonna define the instance type later?

Reply:
Yea i will create a todo for specifying the type for the dashboard instance